lib/LaTeXML/Post.pm

```diff
@@ -447,7 +447,7 @@ sub convertXMTextContent {
   foreach my $node (@nodes) {
     if ($node->nodeType == XML_TEXT_NODE) {
       my $string = $node->textContent;
-      $string =~ s/^\s+/$NBSP/; $string =~ s/\s+$/$NBSP/;    # should we???
```

Comment:
The substitution, or something equivalent, *is* generally needed, at least for MathML.  I don't know why those nbsp should cause problems in sTeX, but I probably *wouldn't* know :> The core of the issue is that sometimes the space matters, such as in
```
$x \mbox{ if } y$
```
and since the leading & trailing whitespace is trimmed in MathML token elements, you need to replace it by nbsp, or find some alternative using mspace or spacing attributes.

Unfortunately, this is all complicated by libxml2's heuristic for adding "ignorable whitespace" when it indents XML output; it doesn't actually consult the schema to determine if whitespace is ignorable or not.  So there's a dance trying to determine whether the whitespace was intended to be ignorable, or whether it needs to be converted. 

Moreover, I recently fixed one case to *not* preserve the spacing, but it seems I've broken a case where it *should* be preserved (& converted).  I'm going to have to collect all my corner cases and try to fix it so it's "right" all (or more) of the time.

But perhaps what you're also saying is that the spaces should *always* be trimmed away and not converted to nbsp in OpenMath?